blueocean-core-js/src/js/i18n/i18n.js

```diff
@@ -130,3 +134,16 @@ export default function i18nTranslator(pluginName, namespace = toDefaultNamespac
 
     return translator;
 }
+
+/**
+ * Create i18n resources bound to a plugin name.
+ * @param pluginName
+ * @param i18nKeys
+ */
+export function mockTranslator(pluginName, i18nKeys = {}) {
+    const cacheKey = buildCacheKey(pluginName);
+    translatorCache[cacheKey] = function mockTranslate(key, opts = {}) {
+        const resolved = i18nKeys[key];
```

Comment:
Used a different approach in commit b4a99ec based on your suggestion: just return the localization key. I think all of our tests want to avoid caring about i18n at all; testing against just the original key value is probably the simplest way to do check that things exist in the UI without having to take into account that a translator is involved.